common/s3util.cpp

```diff
@@ -252,7 +254,10 @@ void S3Util::listObjectsHelper(const string& prefix, const string& delimiter,
     }
   } else {
     if (error_message != nullptr) {
-      *error_message = listObjectResult.GetError().GetMessage();
+      *error_message = folly::sformat("ListObjectsRequest failed with ExceptionName: {}, ErrorMessage: {}, ShouldRetry: {}.", 
+        listObjectResult.GetError().GetExceptionName(), 
+        listObjectResult.GetError().GetMessage(),
+        listObjectResult.GetError().ShouldRetry());
```

Comment:
Can we get the response code: https://sdk.amazonaws.com/cpp/api/LATEST/class_aws_1_1_client_1_1_a_w_s_error.html#adaf553bee7a092a3fb8a1446c1670622

Reply:
Sounds good Just added it

Reply:
Can we also add the request ID? That's what the AWS support team asks for to debug failures.

Reply:
Oh I'm sorry, just read the description. You can ignore my comment :)